File: Leetcode/C++/1. Two Sum/solution.cpp
// https://leetcode.com/problems/two-sum/
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numMap;
        vector<int> answer;
        for (int i = 0; i < nums.size(); i++)
        {
            if (numMap.find(nums[i]) == numMap.end())
            {
                numMap.insert({target - nums[i], i});
            } 
            else {    // Number found
                answer.push_back(numMap[nums[i]]);
                answer.push_back(i);
                return answer;
            }
        }
        return answer;  // Number not found
    }
};

File: Leetcode/Typescript/1. Two Sum/solution.ts
// https://leetcode.com/problems/two-sum/description/

function twoSum(nums: number[], target: number): number[] {
  const index = new Map<number, number>();

  for (let i = 0; i < nums.length; i++) {
    if (index.has(target - nums[i])) {
      return [index.get(target - nums[i]), i];
    }
    index.set(nums[i], i);
  }
}
